### PR TITLE
feat: Nest user and settings under application service

### PR DESCRIPTION
**Pull request by KeithClinard**

### CODE REVIEW

File: addon/components/nrg-appbar.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default class NrgAppbarComponent extends Component {
     const productionEnvironments = this.application.environmentConfig?.[
       'ember-nrg-ui'
     ]?.productionEnvironments ?? ['prod'];
-    const environment = this.application.applicationSettings.localEnvironment;
+    const environment = this.application.settings?.localEnvironment;
```
Comment (TSenter) [inline addon/components/nrg-appbar.js:20]: Can't we guarantee that `this.application.settings` will exist?

**Review by TSenter — CHANGES_REQUESTED** [review]

File: addon/components/nrg-appbar.js (hunk 1 above)
Comment (KeithClinard) [reply addon/components/nrg-appbar.js:20]: Probably, though I don't think it hurts to account for the possibility that it doesn't.

File: tests/integration/components/nrg-home-cards/component-test.js [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
-    this.owner.register('user:current', currentUser, {
-      instantiate: false,
-      singleton: true,
-    });
-    this.owner.inject('service', 'currentUser', 'user:current');
-  });
 
-  test('it renders', async function(assert) {
+  test('it renders', async function (assert) {
+    this.owner.register('service:application', ApplicationServiceStub);
```
[the remark this answers is not in the document]
Comment (KeithClinard) [reply tests/integration/components/nrg-home-cards/component-test.js:20]: I'm not entirely sure why, but it didn't work properly when used in the beforeEach hook.
I'll take another look at it though.

File: addon/services/application.js [hunk 3]
```diff
@@ -21,6 +41,10 @@ export default class ApplicationService extends Service {
   @tracked
   pageTitle = this.environmentConfig?.APP?.name;
 
+  get userRoles() {
+    return this.user?.roles ?? [];
+  }
+
```
[the remark this answers is not in the document]
Comment (KeithClinard) [reply addon/services/application.js:47]: This shouldn't be here.
It's a leftover artifact from my first draft.

File: tests/integration/components/nrg-home-cards/component-test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (KeithClinard) [reply tests/integration/components/nrg-home-cards/component-test.js:20]: Good call.
My issue was likely cause by something else.
I was able to revert this.

**Review by TSenter — APPROVED** [review]

**Pull request merged**